plugins/PrivacyManager/API.php

```diff
@@ -92,9 +94,32 @@ public function findDataSubjects($idSite, $segment)
     {
         Piwik::checkUserHasSomeAdminAccess();
 
+        $siteIds = Site::getIdSitesFromIdSitesString($idSite);
+        $siteIdsWithVisitorLogsOrProfilesEnabled = [];
+        $isLivePluginActivated = Manager::getInstance()->isPluginActivated('Live');
+
+        /*
+         * Only retrieve data from sites that have visitor logs or profiles enabled.
+         * Live::isVisitorProfileEnabled returns false if either logs or profiles
+         * are disabled.
+         */
+        if ($isLivePluginActivated) {
+            foreach ($siteIds as $siteId) {
+                $isVisitorProfileEnabled = Live::isVisitorProfileEnabled($siteId);
+
+                if ($isVisitorProfileEnabled) {
+                    $siteIdsWithVisitorLogsOrProfilesEnabled[] = $siteId;
+                }
+            }
+        }
+
+        if (empty($siteIdsWithVisitorLogsOrProfilesEnabled)) {
+            return [];
+        }
+
```

Comment:
Personally I prefer code with early returns, with less nested code. But it's totally fine to leave it that way

```suggestion
        if (!Manager::getInstance()->isPluginActivated('Live')) {
            return [];
        }

        $siteIds = Site::getIdSitesFromIdSitesString($idSite);
        $siteIdsWithVisitorLogsOrProfilesEnabled = [];

        /*
         * Only retrieve data from sites that have visitor logs or profiles enabled.
         * Live::isVisitorProfileEnabled returns false if either logs or profiles
         * are disabled.
         */
        foreach ($siteIds as $siteId) {
            $isVisitorProfileEnabled = Live::isVisitorProfileEnabled($siteId);

            if ($isVisitorProfileEnabled) {
                $siteIdsWithVisitorLogsOrProfilesEnabled[] = $siteId;
            }
        }

        if (empty($siteIdsWithVisitorLogsOrProfilesEnabled)) {
            return [];
        }

```

Reply:
I totally agree, I also prefer early returns, great suggestion.